src/main/java/com/example/bookstore/repository/book/spec/AuthorSpecificationProvider.java

```diff
@@ -0,0 +1,19 @@
+package com.example.bookstore.repository.book.spec;
+
+import com.example.bookstore.model.Book;
+import com.example.bookstore.repository.SpecificationProvider;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
```

Comment:
this str is duplicated several times. It's probably worth moving it to the constant

Reply:
fix the same issue in all places

Reply:
not fixed, duplicated literal not extracted to constant